annotation/src/main/java/io/smallrye/common/annotation/Experimental.java

```diff
@@ -0,0 +1,24 @@
+package io.smallrye.common.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that specifies that an element is experimental and may change without notice.
+ */
+@Inherited
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE })
```

Comment:
What about some other targets?
`FIELD` potentially makes sense I think (for a CDI producer field for example)
`PACKAGE` too if we add a whole new package somewhere (if it contains a lot of stuff, do we want to separately annotate everything that is contained in that package?)

Reply:
We can include both.

Reply:
Doesn't hurt to support all targets, even if we don't think it necessarily makes sense right now